modules/tinymce/src/plugins/link/main/ts/ui/Controls.ts

```diff
@@ -57,7 +58,14 @@ const setupContextMenu = (editor: Editor): void => {
   const inLink = 'link unlink openlink';
   const noLink = 'link';
   editor.ui.registry.addContextMenu('link', {
-    update: (element) => Utils.hasLinks(editor.dom.getParents(element, 'a') as HTMLAnchorElement[]) ? inLink : noLink
+    update: (element) => {
+      const isInToC = PredicateFind.ancestor(SugarElement.fromDom(element), (ancestor) => Class.has(ancestor, 'mce-toc')).isSome();
```

Comment:
Are there any other classes we should check for? Or perhaps a list of them somewhere we can reuse?

Reply:
Not that I know of, the issue was just with regards to the table of contents. I did consider adding both a list of internal ones ( Which at the moment would just be the one ) and an option to add others by the user, but I thought it'd be best to start off simple. But I can add one or both of those if we want to.